src/agent/service/nodesCrossSignClaim.ts

```diff
@@ -66,7 +67,7 @@ function nodesCrossSignClaim({
         }
         // Verify the claim
         const senderPublicKey = await nodeManager.getPublicKey(
-          payloadData.node1,
+          nodesUtils.decodeNodeId(payloadData.node1)!,
```

Comment:
Is it guaranteed that `payloadData.node1` is a valid and filled node id?

Reply:
It should be valid, yes.

Reply:
@joshuakarp is going to review all of these uses of assertions to catch any bugs in logic.

Reply:
Given that `payloadData` is coming from a decoded claim from some other keynode, I'm not sure that we can assert that the node ID is valid.

The flow is as follows, where keynode `B` is verifying a claim from keynode `A`:
1. `B` receives the claim from `A`
2. `B` attempts to decode the claim (remember, internally it's a JWS with base64 fields) - this also does some structural validation with `ClaimNodeSinglySigned.json`, but does not validate fields
3. `B` attempts to then use the node ID from the decoded claim

So here we actually don't have a guarantee that the node ID is valid (if we assume that a keynode can send a malformed claim).
2. `decodeClaim`

Reply:
The remainder of the usage of node ID in `nodesCrossSignClaim.ts` looks fine to me though.